components/eventing-controller/api/v1alpha1/subscription_conversion.go

```diff
@@ -1,19 +1,340 @@
 package v1alpha1
 
 import (
+	"fmt"
+	"strconv"
+	"strings"
+
 	"github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2"
+	"github.com/pkg/errors"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
+const (
+	hubVersionErrMsg     = "hub version is not the expected v1alpha2 version"
+	multipleSourceErrMsg = "subscription contains more than 1 eventSource"
+)
+
 // ConvertTo converts this Subscription to the Hub version (v2).
-func (s *Subscription) ConvertTo(rawObj conversion.Hub) error {
-	_ = rawObj.(*v1alpha2.Subscription)
+func (src *Subscription) ConvertTo(dstRaw conversion.Hub) error { // nolint
+	dst, ok := dstRaw.(*v1alpha2.Subscription)
+	if !ok {
+		return errors.Errorf(hubVersionErrMsg)
+	}
+	return v1ToV2(src, dst)
+}
+
+// v1ToV2 copies the v1alpha1-type field values into v1alpha2-type field values
+func v1ToV2(src *Subscription, dst *v1alpha2.Subscription) error {
+
+	// ObjectMeta
+	dst.ObjectMeta = src.ObjectMeta
+
+	// SPEC fields
+
+	dst.Spec.ID = src.Spec.ID
+	dst.Spec.Sink = src.Spec.Sink
+	dst.Spec.Source = ""
+
+	src.setV2TypeMatching(dst)
+
+	// protocol fields
+	src.setV2ProtocolFields(dst)
+
+	// Types
+	if err := src.setV2SpecTypes(dst); err != nil {
+		return err
+	}
+
+	// Config
+	src.natsSpecConfigToV2(dst)
+
+	// STATUS Fields
+
+	// Ready
+	dst.Status.Ready = src.Status.Ready
+
+	// Conditions
+	for _, condition := range src.Status.Conditions {
+		dst.Status.Conditions = append(dst.Status.Conditions, ConditionV1ToV2(condition))
+	}
+
+	// event types
+	src.setV2StatusTypes(dst)
+
+	// Backend-specific status
+
+	// BEB
+	src.bebBackendStatusToV2(dst)
+
+	// NATS
+	src.natsBackendStatusToV2(dst)
```

Comment:
Why do we need to copy the status ?  The status does not need to be copied as it will be calculated by the controller. Only the specification is important or am I missing something ?

Reply:
Good point, but I the Status contains some configurations and I wanted to make sure, that if there is no reconciliation(there I am not sure) right after the conversion which would populate the status, that we would have the status there.

Reply:
I don't think an empty status is a problem. This is also the case if we don't have a webhook and the controller is not running.
Do you have an example where the status contains configuration which is required to be copied ?